### PR TITLE
The Start of C Hashtag Support

### PR DESCRIPTION
**Pull request by bscothern-Cricut**

### CODE REVIEW

File: Sources/FishyJoesCSharpRuntime/File.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//
+//  File.swift
```
Comment (cobbal) [inline Sources/FishyJoesCSharpRuntime/File.swift]: File.swift

File: Sources/FishyJoesCore/FishyJoesContext.swift [hunk 2]
```diff
@@ -12,10 +12,22 @@ public class FishyJoesContext {
...
+        NodeTranslator.self,
+        KotlinTranslor.self,
+        CSharpTranslator.self,
+    ]
+    
+    lazy var translators: [Translator] = translatorTypes.map({ $0.init() })
+
+    let nodeTranslator = NodeTranslator()
+    let kotlinTranslator = KotlinTranslor()
+    let cSharpTranslator = CSharpTranslator()
```
Comment (cobbal) [inline Sources/FishyJoesCore/FishyJoesContext.swift:23]: I don't like the duplicate translators.

File: Sources/FishyJoesCore/FishyJoesContext.swift [hunk 3]
```diff
@@ -121,27 +135,47 @@ public class FishyJoesContext {
...
+    ///
+    /// - Parameters:
+    ///   - rootClass: The root to put all nested classes inside.
+    ///   - classes: The classes to process.
+    ///   - seperator: The separator in the name to split on for namespaces.
+    /// - Returns: The resulting fragments with their inner classes properly processed.
+    func processInnerClasses<C: Class>(
+        rootClass: @autoclosure () -> C,
+        in classes: inout [C],
+        seperator: Character = "."
```
Comment (cobbal) [inline Sources/FishyJoesCore/FishyJoesContext.swift]: typo: separator.

File: Sources/FishyJoesCore/FishyJoesContext.swift [hunk 4]
```diff
@@ -121,27 +135,47 @@ public class FishyJoesContext {
...
 
             var containingClass = rootClass
             while let outer = namespace.popLast() {
                 guard let next = containingClass.innerClasses.first(where: { $0.unqualifiedName == outer }) else {
-                    fatalErr("""
-                        while processing \(ktClass.name):
-                        Unable to find class \(outer) in class \(containingClass.name):
-                        \(containingClass.innerClasses.map(\.name))
-                        """)
+                    fatalError()
```
Comment (cobbal) [inline Sources/FishyJoesCore/FishyJoesContext.swift]: The error message is important, as users may have forgotten an annotation

File: Sources/FishyJoesCore/Unparse/Class.swift [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/// The FishyJoes Class properties that are needed for processing of fragments.
+protocol Class: _Class where InnerClass == Self {}
```
Comment (cobbal) [inline Sources/FishyJoesCore/Unparse/Class.swift]: I hate this protocol name.

Comment (cobbal) [reply Sources/FishyJoesCore/Unparse/Class.swift]: Maybe "NestedClass" would be better.

File: Sources/FishyJoesExecute/CodeGen.swift [hunk 6]
```diff
@@ -32,19 +33,22 @@ struct CodeGen: ParsableCommand {
     @Flag(name: .long, inversion: .prefixedNo, help: "Generate a Kotlin package without android support (much faster)")
     var kotlinFast: Bool = false
 
+    @Flag(name: .customLong("C#"), inversion: .prefixedNo, help: "Generate a C# Package")
```
Comment (cobbal) [inline Sources/FishyJoesExecute/CodeGen.swift]: "#" is usually a shell comment character, so I don't think we should use it in an option.

File: Sources/FishyJoesExecute/Platform.swift [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+            return ".build/x86_64-unknown-linux-gnu/release"
+            #else
+            fatalError("unknown host OS")
+            #endif
+        case .kotlinAndroid(.arm):
+            return ".build/android-build/armv7-none-linux-androideabi/release"
+        case .kotlinAndroid(let arch):
+            return ".build/android-build/\(arch.rawValue)-unknown-linux-android/release"
+        case .cSharp:
+            var cSharpPlatformBuildDirectory: String {
```
Comment (cobbal) [inline Sources/FishyJoesExecute/Platform.swift:132]: You can do local getters? nifty.

**Review by cobbal — CHANGES_REQUESTED** [review]
Looks pretty good overall. Some good cleanup in there. Few minor issues.

File: Sources/FishyJoesCore/FishyJoesContext.swift (hunk 2 above)
Comment (cobbal) [reply Sources/FishyJoesCore/FishyJoesContext.swift:23]: In fact, we should probably just create the translators up front, drop the init requirement, then stick them in an array.

Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/FishyJoesContext.swift:23]: Test comment from Xcode

Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/FishyJoesContext.swift:23]: Fixed.

File: Sources/FishyJoesCore/FishyJoesContext.swift (hunk 3 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/FishyJoesContext.swift]: Fixed.

File: Sources/FishyJoesCore/FishyJoesContext.swift (hunk 4 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/FishyJoesContext.swift]: Fixed. Don't know how I lost that.

File: Sources/FishyJoesCore/Unparse/Class.swift (hunk 5 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/Unparse/Class.swift]: Fixed.

File: Sources/FishyJoesExecute/CodeGen.swift (hunk 6 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesExecute/CodeGen.swift]: Fixed.

File: Sources/FishyJoesExecute/Platform.swift (hunk 7 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesExecute/Platform.swift:132]: Yep, they are great.

**Review by cobbal — APPROVED** [review]
Looks good.

**Pull request merged**